### PR TITLE
[Hotfix] /4.0.1 NFT improvement

### PR DESCRIPTION
**Pull request by sethkfman**
**Description**

_Write a short description of the changes included in this pull request._

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**

Resolves #???

### CODE REVIEW

**Review by gantunesr — COMMENTED** [review]
LGTM. I'll give it another look tomorrow morning just in case I missed something

**Review by cortisiko — CHANGES_REQUESTED** [review]
**Issue 1.**
With NFT auto detection turned off when i try to manually import my NFT, nothing happens. http://recordit.co/yLvEmxe7pP

to reproduce:
log in to your wallet
tap on the NFT tab
do not turn on NFT token detection but toggle on “enable OpenSea API”.
 Import an NFT in your wallet
notice the NFT does not import. 
I killed the app and relaunch and the NFT does not appear. 

second set of reproduction steps if the above does not work:
log in to your wallet
tap on the NFT tab
toggle on NFT token detection 
remove one of your NFT’s. 
toggle token detection off. 
import the NFT you removed. 
notice the NFT does not import. 


**Issue 2.**
 The NFT auto-detection message appears on the import tokens view: https://www.screencast.com/t/2Fzobx9l4gql

**Issue 3.** (being nit picky here.)
going to the NFT detection toggle on the privacy & settings view is not as fluid compared to going to the enhance detection toggle on the experimental view.

NFT detection toggle: http://recordit.co/5InmXTVJmS
Token detection toggle: http://recordit.co/Q3qj6Fhzrv


**Potential Issue 4.** 
This happened to me a couple times when i first started testing. Now I can no longer reproduce. Can you look into it and ensure it is not an issue: 

 _when NFT auto detection is switched on, it only imports your first (oldest) NFT. The other NFT's do not import._ 

to reproduce: 
log in to your wallet
tap on the NFT tab
 turn on NFT token detection. Notice your first NFT imports
notice the other NFT’s does not import. 

unfortunately I cannot provide a recording. But it did happen.

File: app/components/Views/AddAsset/index.js [hunk 1]
```diff
@@ -76,15 +87,28 @@ class AddAsset extends PureComponent {
...
 		} = this.props;
+		const { dismissNftInfo } = this.state;
+
 		return (
 			<SafeAreaView style={styles.wrapper} testID={`add-${assetType}-screen`}>
+				{isMainNet(chainId) && !dismissNftInfo && !useCollectibleDetection && (
+					<View style={styles.infoWrapper}>
+						<CollectibleDetectionModal onDismiss={this.dismissNftInfo} navigation={navigation} />
+					</View>
+				)}
```
Comment (gantunesr) [inline app/components/Views/AddAsset/index.js:111]: This is one of the bugs reported. It's needed to have a conditional as the one below.

```javascript
{assetType === 'token' ? (
  //...
) : (
  //...
)}
```

**Review by gantunesr — COMMENTED** [review]
Left a small question

File: app/components/Views/Settings/SecuritySettings/index.js [hunk 2]
```diff
@@ -321,6 +340,8 @@ class Settings extends PureComponent {
 				hintText: manualBackup,
 			});
 		}
+
+		if (this.props.route?.params?.scrollToBottom) this.scrollView?.scrollToEnd({ animated: false });
```
[the remark this answers is not in the document]
Comment (sethkfman) [reply app/components/Views/Settings/SecuritySettings/index.js:344]: @gantunesr Yes, if we add a lot more settings these toggles could be bumped higher. I was having issues finding away to scroll to a specific component on the view.

**Review by cortisiko — APPROVED** [review]
🌮 🌮 🌮

**Review by gantunesr — APPROVED** [review]
LGTM

**Pull request merged**